src/crate/testing/layer.py

```diff
@@ -40,34 +42,52 @@ def __init__(self,
         :param cluster_name: the name of the cluster to join/build. Will be
                              generated automatically if omitted.
         :param host: the host to bind to. defaults to 'localhost'
+        :param settings: further settings that do not deserve a keyword argument
+                         will be prefixed with ``es.``.
         """
         self.keepRunning = keepRunning
         crate_home = os.path.abspath(crate_home)
-        servers = ['%s:%s' % (host, port)]
-        self.crate_servers = ['http://%s:%s' % (host, port)]
         if crate_exec is None:
             crate_exec = os.path.join(crate_home, 'bin', 'crate')
         if crate_config is None:
             crate_config = os.path.join(crate_home, 'config', 'crate.yml')
         if cluster_name is None:
             cluster_name = "Testing{0}".format(port)
-        start_cmd = (
-            crate_exec,
-            '-Des.index.storage.type=memory',
-            '-Des.node.name=%s' % name,
-            '-Des.cluster.name=%s' % cluster_name,
-            '-Des.http.port=%s-%s' % (port, port),
-            '-Des.network.host=%s' % host,
-            '-Des.discovery.type=zen',
-            '-Des.discovery.zen.ping.multicast.enabled=%s' % ("true" if multicast else "false"),
-            '-Des.config=%s' % crate_config,
-            '-Des.path.conf=%s' % os.path.dirname(crate_config),
-        )
-        if transport_port:
-            start_cmd += ('-Des.transport.tcp.port=%s' % transport_port,)
-        super(CrateLayer, self).__init__(name, servers=servers, start_cmd=start_cmd)
+        settings = self.create_settings(crate_config, cluster_name, name, host, port, transport_port, multicast, settings)
+        start_cmd = (crate_exec, ) + tuple(["-Des.%s=%s" % (setting, value) for setting, value in settings.items()])
```

Comment:
`settings.items()` returns a tuple that can be used for string formatting directly
e.g. `start_cmd = (crate_exec, ) + tuple(["-Des.%s=%s" % opt for opt in settings.items()])`